Mathlib/Analysis/InnerProductSpace/Adjoint.lean

```diff
@@ -506,6 +506,91 @@ theorem im_inner_adjoint_mul_self_eq_zero (T : E →ₗ[𝕜] E) (x : E) :
 
 end LinearMap
 
+section Unitary
+
+variable {H : Type*} [NormedAddCommGroup H] [InnerProductSpace 𝕜 H] [CompleteSpace H]
+
+namespace ContinuousLinearMap
+
+variable {K : Type*} [NormedAddCommGroup K] [InnerProductSpace 𝕜 K] [CompleteSpace K]
+
+theorem inner_map_map_iff_star_mul_self (u : H →L[𝕜] K) :
+    (∀ x y : H, ⟪u x, u y⟫_𝕜 = ⟪x, y⟫_𝕜) ↔ adjoint u ∘L u = 1 := by
+  refine ⟨fun h ↦ ext fun x ↦ ?_, fun h ↦ ?_⟩
+  · refine ext_inner_right 𝕜 fun y ↦ ?_
+    simpa [star_eq_adjoint, adjoint_inner_left] using h x y
+  · simp [← adjoint_inner_left, ← comp_apply, h]
+
+theorem norm_map_iff_star_mul_self (u : H →L[𝕜] K) :
+    (∀ x : H, ‖u x‖ = ‖x‖) ↔ adjoint u ∘L u = 1 := by
+  rw [LinearMap.norm_map_iff_inner_map_map u, u.inner_map_map_iff_star_mul_self]
+
+theorem norm_map_of_mem_unitary
+    (u : H →L[𝕜] H) (hu : u ∈ unitary (H →L[𝕜] H)) (x : H) : ‖u x‖ = ‖x‖ :=
+  u.norm_map_iff_star_mul_self.mpr (unitary.star_mul_self_of_mem hu) x
+
+theorem inner_map_map_of_mem_unitary
+    (u : H →L[𝕜] H) (hu : u ∈ unitary (H →L[𝕜] H)) (x y : H) : ⟪u x, u y⟫_𝕜 = ⟪x, y⟫_𝕜 :=
+  u.inner_map_map_iff_star_mul_self.mpr (unitary.star_mul_self_of_mem hu) x y
+
+end ContinuousLinearMap
+
+namespace unitary
+
+theorem norm_map (u : unitary (H →L[𝕜] H)) (x : H) : ‖u.val x‖ = ‖x‖ :=
```

Comment:
```suggestion
theorem norm_map (u : unitary (H →L[𝕜] H)) (x : H) : ‖(u : H →L[𝕜] H) x‖ = ‖x‖ :=
```
I think we prefer the coercion over the explicit `.val` (although in the end they are the same thing). Same thing several times below.

Reply:
I'm not entirely sure how I feel about this as a prescriptive statement, but I've implemented it nevertheless.